sympy/core/numbers.py

```diff
@@ -1934,7 +1939,8 @@ def _Rrel(self, other, attr):
         try:
             other = _sympify(other)
         except SympifyError:
-            return NotImplemented
+            from .relational import InvalidComparison
+            raise InvalidComparison("Invalid comparison %s > %s" % (self, other))
```

Comment:
I don't think this should be changed. If the other operand can not be sympified then `NotImplemented` should be returned to allow cooperative dispatch. If both methods return NotImplemented then TypeError is correct.

The point of having `InvalidComparison` is so that we can catch it but also avoid catching other kinds of TypeError. This one should just be a TypeError.

Reply:
This used to throw a `TypeError` saying the same thing, so it is simply the status of the code two years ago but with TypeError replaced with InvalidComparison.